theurgy.lic

```diff
@@ -55,75 +130,357 @@ def buy_wine
 
-until ['flawless performance to those on high', 'In your condition'].include? DRC.bput('dance', 'flawless performance to those on high', 'You begin to dance', 'Your actions grow', 'Your dance', 'but you falt', 'In your condition')
+recite_prayer = proc do
+  fput "recite Meraud, power the holy fires that unleash my righteous vengeance;Chadatru, guide my sword to swing in justice;Everild, give me the power to conquer my enemies;Truffenyi, let me not lose sight of compassion and mercy;Else, I will become like those I despise;Urrem'tier, receive into your fetid grasp these wicked souls;May the Tamsine's realms never know their evil ways again;May all the Immortals guide your faithful soldier #{checkname}."
   pause 1
   waitrt?
-  DRC.fix_standing
 end
 
-fput "get #{$settings.flint_lighter}", 'You get a', 'You are already'
-get_from_container('incense')
-while 'nothing happens' == DRC.bput('light my incense with flint', 'nothing happens', 'bursts into flames')
+reflectory = proc do
+  DRCT.walk_to '5988'
+  fput 'meditate'
   waitrt?
 end
+
+invoke_parchment = proc do
+  DRCT.walk_to('1230')
+  if 'but there is nothing in there like that' != DRC.bput("rummage /C parchment my #{$settings.theurgy_supply_container}", 'but there is nothing in there like that', 'pale golden prayer parchment')
+    get_from_container('golden parchment')
+    fput 'invoke my parchment'
+    waitfor 'You conclude'
+    put_in_container('golden parchment')
+  end
+end
+
+######################## CUSTOM TRAVEL ########################
+
+enter_altar = proc do
+  DRCT.walk_to('1164')
+  move 'go shack'
+  move 'go hole'
+  move 'west'
+  move 'up'
+  waitrt?
+  move 'go path'
+end
+
+exit_altar = proc do
+  move 'go path'
+  move 'go hole'
+  waitrt?
+  move 'east'
+  move 'go hole'
+  move 'out'
+end
+
+######################## MISC ########################
+
+def research_check
+  case find_current_project = DRC.bput("research status", / (\d+)% complete with a portion .*\./, 'not researching anything', 'project about')
+  when /not researching anything/
+    return
+  when /project about/
+    research_type = ["Fundamental", "Augmentation", "Stream", "Sorcery", "Utility", "Warding"]
+    $hold_research = research_type.find{ |$hold_research| find_current_project =~ /#{$hold_research}/i }
+    if $hold_research.nil?
+      return
+    end
+  when /complete with a portion/
+    echo find_current_project
+    research_type = ["Fundamental", "Augmentation", "Stream", "Sorcery", "Utility", "Warding"]
+    $hold_research = research_type.find{ |$hold_research| find_current_project =~ /#{$hold_research}/i }
+    if $hold_research.nil?
+      return
+    end
+    if find_current_project[1,2].to_i >= 60
+      wait_time = (101 - find_current_project[1,2].to_i) * 3
+      Flags.reset('research_done')
+      pause (wait_time)
+    else
+      echo "stopping research"
+    end
+    fput 'prep bless'
+  end
+  echo $hold_research
+  return $hold_research
+end
+
+def controller(method)
```

Comment:
And change the arg named method to prevent confusion with the ruby method named method.